src/Microsoft.ML.Data/DataView/ArrayDataViewBuilder.cs

```diff
@@ -78,13 +78,27 @@ public void AddColumn<T>(string name, PrimitiveType type, params T[] values)
         public void AddColumn(string name, ValueGetter<VBuffer<ReadOnlyMemory<char>>> getKeyValues, ulong keyMin, int keyCount, params uint[] values)
         {
             _host.CheckValue(getKeyValues, nameof(getKeyValues));
-            _host.CheckParam(keyCount > 0, nameof(keyCount));
+            //_host.CheckParam(keyCount > 0, nameof(keyCount));
```

Comment:
>//_host.CheckParam(keyCount > 0, nameof(keyCount)); [](start = 12, length = 51)

khm

Reply:
I dont know what khm means, but you can pass 0 for count. However I will revert this change.

Count is confusing for this api -- there is Count in terms of Keys, meaning that if I have 0,2,5, my Count should be 5 (or maybe 6 since everything is shifted by 1?). However  Count in this case should be 3 because we are writing three items. However you can make it 0 (say if the count is greater than int max) and then it works  more like you would expect in the former case as the saving does work AND I get a schema of Key<min-*>. 

Talking with you and Pete, Count in this case should reflect how many items are being written, so I will add a comment to reflect this but it does feel like there is a bug here.

---
In reply to: [243448353](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243448353) [](ancestors = 243448353)

Reply:
Khm is just Hmmmm but with Russian accent. :)
I just saw what you comment something, and in general it's not best practice. Either remove it completely  or leave it as is without commenting.



---
In reply to: [243452554](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243452554) [](ancestors = 243452554,243448353)